File: solutions/2785. Sort Vowels in a String/2785_1.cpp
// Time complexity: O(sort)
// Space complexity: O(n)
class Solution {
 public:
  string sortVowels(const string& s) {
    vector<char> vowels;
    for (const char c : s)
      if (isVowel(c))
        vowels.push_back(c);

    ranges::sort(vowels);

    string ans;
    int i = 0;
    for (const char c : s)
      ans += isVowel(c) ? vowels[i++] : c;

    return ans;
  }

 private:
  bool isVowel(char c) {
    static constexpr string_view kVowels = "aeiouAEIOU";
    return kVowels.find(c) != string_view::npos;
  }
};

File: solutions/2785. Sort Vowels in a String/2785_1.py
# Time complexity: O(sort)
# Space complexity: O(n)
class Solution:
  def sortVowels(self, s: str) -> str:
    kVowels = 'aeiouAEIOU'
    ans = []
    vowels = sorted([c for c in s if c in kVowels])

    i = 0
    for c in s:
      if c in kVowels:
        ans.append(vowels[i])
        i += 1
      else:
        ans.append(c)

    return ''.join(ans)      
